lib/model/sentdownloadstate.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2015 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package model
+
+import (
+	"time"
+
+	"github.com/syncthing/protocol"
+	"github.com/syncthing/syncthing/lib/scanner"
+)
+
+// sentFolderDownloadState represents a state of what we've announced as available
+// to some remote device for a specific folder.
+type sentFolderDownloadState struct {
+	fileBlocks  map[string][]protocol.BlockInfo
```

Comment:
This is kind of heavy. Could we not get away with just storing the highest block index we've sent or something like that?